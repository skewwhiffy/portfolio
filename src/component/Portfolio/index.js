import React from 'react'

const Portfolio = () => {
  const sites = [{
    name: 'Twitter',
    url: 'https://twitter.com',
    github: 'https://github.com/twitter',
    screenshot: 'twitter.png'
  }, {
    name: 'XKCD',
    url: 'https://xkcd.com',
    github: 'https://github.com/xkcd',
    screenshot: 'xkcd.png'
  }, {
    name: 'Signal',
    url: 'https://signal.org',
    github: 'https://github.com/signalapp',
    screenshot: 'signal.png'
  }, {
    name: 'OnePlus',
    url: 'https://www.oneplus.com',
    github: 'https://github.com/OnePlusOSS',
    screenshot: 'oneplus.png'
  }, {
    name: 'QMK Keyboards',
    url: 'https://qmk.fm',
    github: 'https://github.com/qmk',
    screenshot: 'qmk.png'
  }, {
    name: 'OLKB Keyboards',
    url: 'https://olkb.com/',
    github: 'https://github.com/olkb',
    screenshot: 'olkb.png'
  }]
  return (
    <div className="portfolio">
      <h1>Portfolio</h1>
      <p>
        <b>The following information is not real! Screenshots and URLs are for demonstration purposes only!</b>
      </p>
      {
        sites.map(it => {
          return (
            <div>
              <h2>{it.name}</h2>
              <p>
                <img
                  className="screenshot"
                  src={require(`../../asset/portfolio/${it.screenshot}`).default}
                  alt={`${it.name} screenshot`}
                />
                <ul>
                  <li>URL: <a href={it.url}>{it.url}</a></li>
                  <li>Github: <a href={it.github}>{it.github}</a></li>
                </ul>
              </p>
              <br />
            </div>
          )
        })
      }
    </div>
  )
}

export default Portfolio