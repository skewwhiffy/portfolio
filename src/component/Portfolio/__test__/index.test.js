import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Portfolio from '..';

afterEach(cleanup)

describe('Portfolio component', () => {
  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<Portfolio />)

    expect(asFragment()).toMatchSnapshot();
  });
})