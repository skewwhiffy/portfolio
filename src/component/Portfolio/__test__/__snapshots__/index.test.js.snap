// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="portfolio"
  >
    <h1>
      Portfolio
    </h1>
    <p>
      <b>
        The following information is not real! Screenshots and URLs are for demonstration purposes only!
      </b>
    </p>
    <div>
      <h2>
        Twitter
      </h2>
      <p>
        <img
          alt="Twitter screenshot"
          class="screenshot"
        />
      </p>
      <ul>
        <li>
          URL: 
          <a
            href="https://twitter.com"
          >
            https://twitter.com
          </a>
        </li>
        <li>
          Github: 
          <a
            href="https://github.com/twitter"
          >
            https://github.com/twitter
          </a>
        </li>
      </ul>
      <br />
    </div>
    <div>
      <h2>
        XKCD
      </h2>
      <p>
        <img
          alt="XKCD screenshot"
          class="screenshot"
        />
      </p>
      <ul>
        <li>
          URL: 
          <a
            href="https://xkcd.com"
          >
            https://xkcd.com
          </a>
        </li>
        <li>
          Github: 
          <a
            href="https://github.com/xkcd"
          >
            https://github.com/xkcd
          </a>
        </li>
      </ul>
      <br />
    </div>
    <div>
      <h2>
        Signal
      </h2>
      <p>
        <img
          alt="Signal screenshot"
          class="screenshot"
        />
      </p>
      <ul>
        <li>
          URL: 
          <a
            href="https://signal.org"
          >
            https://signal.org
          </a>
        </li>
        <li>
          Github: 
          <a
            href="https://github.com/signalapp"
          >
            https://github.com/signalapp
          </a>
        </li>
      </ul>
      <br />
    </div>
    <div>
      <h2>
        OnePlus
      </h2>
      <p>
        <img
          alt="OnePlus screenshot"
          class="screenshot"
        />
      </p>
      <ul>
        <li>
          URL: 
          <a
            href="https://www.oneplus.com"
          >
            https://www.oneplus.com
          </a>
        </li>
        <li>
          Github: 
          <a
            href="https://github.com/OnePlusOSS"
          >
            https://github.com/OnePlusOSS
          </a>
        </li>
      </ul>
      <br />
    </div>
    <div>
      <h2>
        QMK Keyboards
      </h2>
      <p>
        <img
          alt="QMK Keyboards screenshot"
          class="screenshot"
        />
      </p>
      <ul>
        <li>
          URL: 
          <a
            href="https://qmk.fm"
          >
            https://qmk.fm
          </a>
        </li>
        <li>
          Github: 
          <a
            href="https://github.com/qmk"
          >
            https://github.com/qmk
          </a>
        </li>
      </ul>
      <br />
    </div>
    <div>
      <h2>
        OLKB Keyboards
      </h2>
      <p>
        <img
          alt="OLKB Keyboards screenshot"
          class="screenshot"
        />
      </p>
      <ul>
        <li>
          URL: 
          <a
            href="https://olkb.com/"
          >
            https://olkb.com/
          </a>
        </li>
        <li>
          Github: 
          <a
            href="https://github.com/olkb"
          >
            https://github.com/olkb
          </a>
        </li>
      </ul>
      <br />
    </div>
  </div>
</DocumentFragment>
`;
